core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -623,10 +623,14 @@ static void check_rcc_secure_configuration(void)
 	bool mckprot = stm32_rcc_is_mckprot();
 	enum stm32mp_shres id = STM32MP1_SHRES_COUNT;
 	bool have_error = false;
+	uint32_t state = 0;
 
-	if (stm32mp_is_closed_device() && !secure)
+	if (stm32_bsec_get_state(&state))
 		panic();
 
+	if (state == BSEC_STATE_SEC_CLOSED && !secure)
+		panic("Closed device mandates secure RCC");
```

Comment:
@GseoC tbh, I see one inconvenience related to this particular check.

Closed board requires changing compatible for RCC `s/st,stm32mp1-rcc/st,stm32mp1-rcc-secure/g` in device tree , which basically does not allow to use the same DTS (same relates to TF-A) on both open/closed boards.

Does it make sense to force `enable_rcc_tzen()` if the board is closed, instead if checking for correct compatible and panic() if it's not `st,stm32mp1-rcc-secure`?

Reply:
Hello Igor,

This changes does not imply a functionnal change, I just use another API

`if (stm32mp_is_closed_device() && !secure)` => `(state == BSEC_STATE_SEC_CLOSED && !secure)`

Reply:
@igoropaniuk ,

Are you OK with the change?

Reply:
> This changes does not imply a functionnal change, I just use another API
Sorry for the confusion. Just to clarify, I didn't mean this particular line but rather a whole approach regarding these security checks, where I have to adjust a device tree for a closed board to comply them. Enforcing hardening of internal peripherals, like RCC, instead of checking if it was explicitly configured to secure in the device tree (`s/st,stm32mp1-rcc/st,stm32mp1-rcc-secure/g`) might be a better option.

@etienne-lms @GseoC I know that it might be out of this PR scope discussion, but have you ever considered such approach? IMHO having two separate device trees for open/closed boards introduces a lot of inconviniece.

Reply:
@igoropaniuk , just to clarify what is implemented:

`which basically does not allow to use the same DTS (same relates to TF-A) on both open/closed boards.`
 It is possible to share a device-tree between a closed/open board. What is at stake here is the "hosting" of the RCC peripheral. It is possible to have a RCC secure on an open board. However, having a closed board using OP-TEE mandates a secure RCC and the use of SCMI.

Enforcing of secure RCC is done in stm32mp1_clock_provider_probe(), depending on the RCC compatible: 
```
	if (compat_data == &non_secure_rcc)
		disable_rcc_tzen();
	else
		enable_rcc_tzen();
```

If you have to change your device-tree from an open board to a closed board, that means that you're using a non-secure RCC on your open board, right?

Reply:
@GseoC thanks for the explaition!

Reply:
You're welcome